src/components/search/SearchBar.vue

```diff
@@ -0,0 +1,82 @@
+<template>
+  <div class="search-box-wrapper">
+    <input
+      id="search-box"
+      v-model="searchQuery"
+      :placeholder="placeholder"
+      @keyup.enter="search(searchQuery)"
+    />
+    <div :class="searchQuery ? 'hidden' : 'visible'">
+      <img
+        svg-inline
+        class="icon icon-search"
+        alt="icon-search"
+        src="@/assets/icons/search.svg"
+      />
+    </div>
+    <div
+      :class="searchQuery ? 'visible' : 'hidden'"
+      @click="searchQuery = null"
+    >
+      <img
+        svg-inline
+        class="icon icon-close"
+        alt="icon-close"
+        src="@/assets/icons/close.svg"
+      />
+    </div>
+  </div>
+</template>
+<script>
+import { ref } from "vue";
+export default {
+  name: "SearchBar",
+  props: {
+    query: {
+      default: "",
+      type: String,
+    },
+    placeholder: {
+      type: String,
+      default: "",
+    },
+  },
+  setup(props, context) {
+    const searchQuery = ref(null);
```

Comment:
We don't need a ref here. We emit an event with a value and we don't care what happens next. Don't see value in making it reactive

Reply:
yeah that is right but if I do like this instead, then it is needed - and this is probably better this way than calling it with an argument from the template
```
const search = () => {
      if (!searchQuery.value) return;
      context.emit("query", searchQuery.value);
    };
```